src/client.ts

```diff
@@ -2535,32 +2523,68 @@ export class StreamChat<
     } else if (timeoutOrExpirationDate instanceof Date) {
       pinExpires = timeoutOrExpirationDate.toISOString();
     }
-    return this.updateMessage({
-      ...message,
-      pinned: true,
-      pin_expires: pinExpires,
-    });
+    return pinExpires;
+  }
+
+  /**
+   * _messageId - extracts string message id from either message object or message id
+   * @param {string | { id: string }} messageOrMessageId message object or message id
+   * @param {string} caller caller function name to report an error in case of message id absence
+   */
+  _messageId(messageOrMessageId: string | { id: string }, caller: string) {
```

Comment:
how about 
```
_validateAndGetMessageId(messageOrMessageId, "Please specify the message id when calling pinMessage")
```

Reply:
`_messageId()` doesn't really say much about the function!!

Reply:
Agreed, going to change that